2009-flu-uk.qmd

```diff
@@ -1 +1,437 @@
+# Third argument -> Proportion of infection detected 
+par0 <- c(5e-5, 1.3, 0.1, 10)
+
+# This vector should be passed as a list
+dust_system_update_pars(sys = mod,
+                        pars = list(alpha=par0[1],
+                                    R_0=par0[2],
+                                    h_times = hol_t,
+                                    h_values= hol_v))
+```
+
+Then the model can be run by specifying the time points between which to integrate the ODEs. Note that the first point (here 0) set the inital time of the integration. Once the time vector t is defined we can run the model by using the "$run(t)" method.
+```{r}
+t <- c(0,data$Days)
+dust_system_set_state_initial(mod)
+y <- dust_system_simulate(mod, t)
+```
+
+Let's plot the I compartment over time.
+
+```{r}
+plot(t, dust_unpack_state(mod, y)$incidence, type = "l", col = "red", lwd = 2)
+```
+
+It looks a bit like the actual epidemic; a good sign, but let's compare our model with the data.
+
+```{r}
+plot(t, dust_unpack_state(mod, y)$incidence*par0[3], type = "l", col = "red", lwd = 2)
+points(data$Days, data$Cases)
+```
+
+We seem to be overpredicted the expected number of cases by a factor two or so, but the MCMC pipeline should allow us to infer the distribution of the parameters consistent with the observations.
+
+## The MCMC pipeline
+
+We now have a dataset and a working model, all loaded in our R environment. We need to link the output of this model with the data in order to derive a likelihood function. As we are working within a Bayesian framework we also need to define the prior distribution for our parameters. 
+
+### Setting up the likelihood
+
+The likelihood is a function that takes a parameter as argument and return the probability density that our model (transmission model + observation) generates exactly the observed data. This number is usually very small as it is the product of each individual observation density given the model. As opposed to the example in the lecture, we thus always work using the log-density values rather than the density. Products then become sums and divisions become differences.
+
+```{r}
+data$time <- data$Days
+filter <- dust_unfilter_create(seir, data = data, time_start = 0)
+dust_likelihood_run(filter, list(alpha=par0[1],
+                                    R_0=par0[2],
+                                    h_times = hol_t,
+                                    h_values= hol_v))
+```
+
+This is the likelihood associated with the dust model written using odin. We know need to move this into the monty statistical world to be able to integrate this into our pipeline. For this use the `packer` concept that allows us to definne a one-to-one translation between the two worlds.
+
+```{r}
+packer <- monty_packer(c("alpha", "R_0", "rho", "eta"),
+                       fixed = list(h_times = hol_t, h_values= hol_v))
+```
+
+This defines two functions that allows to 'pack' (define) and 'unpack' (define) our parameters.
```

Comment:
This has been explained by this point and we should just link back to it

Reply:
This has been removed and a ref to the relevant section added.